### ISSUE TITLE
Why does --create-new-ontology take an argument?

### ISSUE THREAD

**Issue by jamesaoverton** [ontodev/robot#132]
Hey @cmungall: I got confused when I tried this for the first time today. The `robot reason --help` says:

    -n,--create-new-ontology <arg>     switch to a new ontology containing only the inferences

I tried making `<arg>` a filename, but that didn't work. It worked when I set `<arg>` to `true`. But I don't expect a Boolean flag to take an argument -- it's either present or it's not.

**Comment by jamesaoverton** [comment]
Same goes for ` -s,--remove-redundant-subclass-axioms <arg>`, I think.

**Comment by jamesaoverton** [comment]
No, `-s` is true by default, so it's not the same.

**Comment by cmungall** [comment]
Re: file vs boolean. I agree it's confusing.

Re: boolean args. I thought this was the way things worked in Robot. I'm sure I was modeling it on existing code, but I may have messed up...

**Comment by cmungall** [comment]
Yes, I took the `-s` option as my starting point: https://github.com/ontodev/robot/blame/01ac3261bf27e0a7869ae106fd2fa1642bd36391/robot-core/src/main/java/org/obolibrary/robot/ReasonOperation.java#L104-L107

So is the idea:

 - booleans that are default false should have a zero-arg option
 - booleans that are default true should have a 1-arg option, with a silent `--FOO true`

**Comment by jamesaoverton** [comment]
Part of the problem is that Apache Commons CLI calls the argument `<arg>` by default. We can use `setArgName("true")` to make this more clear. It's a little awkward, so maybe a helper function would be good. That change should be made to all commands.

I think the better approach would be for all boolean flags to be false when not present, true when present. I messed this up back when I added `--remove-redundant-subclass-axioms` defaulting to true -- it should have been `--keep-redundant-subclass-axioms` defaulting to false.

I don't want to make a breaking change. It might be possible to use `setOptionalArg(true)` to grandfather these cases. I'll think about it some more.
